extension/src/ohlc.rs

```diff
@@ -14,65 +15,164 @@ use tspoint::TSPoint;
 pub mod toolkit_experimental {
     use super::*;
 
+    flat_serialize_macro::flat_serialize! {
```

Comment:
Would appreciate some review on this specifically. Was kinda winging it with getting `flat_serialize!`ing an `enum` to work, and this arrangement was the first one the compiler didn't complain about.

Reply:
I haven't used our flat_serialize crate before but this seems like it would be fine to me.